plugins/async-node/core/src/index.test.ts

```diff
@@ -1,42 +1,134 @@
 import { expect, test } from "vitest";
-import type { Node, InProgressState } from "@player-ui/player";
-import { Player } from "@player-ui/player";
-import { waitFor } from "@testing-library/react";
-import { AsyncNodePlugin, AsyncNodePluginPlugin } from "./index";
+import { Node, InProgressState, Resolver } from '@player-ui/player';
+import { Player } from '@player-ui/player';
+import { waitFor } from '@testing-library/react';
+import { AsyncNodePlugin, AsyncNodePluginPlugin } from './index';
 
 const basicFRFWithActions = {
-  id: "test-flow",
+  id: 'test-flow',
   views: [
     {
-      id: "my-view",
+      id: 'my-view',
       actions: [
         {
           asset: {
-            id: "action-0",
-            type: "action",
-            value: "{{foo.bar}}",
-          },
+            id: 'action-0',
+            type: 'action',
+            value: '{{foo.bar}}'
+          }
         },
         {
-          id: "uhh",
-          async: "true",
-        },
-      ],
-    },
+          id: 'uhh',
+          async: 'true'
+        }
+      ]
+    }
   ],
   navigation: {
-    BEGIN: "FLOW_1",
+    BEGIN: 'FLOW_1',
     FLOW_1: {
-      startState: "VIEW_1",
+      startState: 'VIEW_1',
       VIEW_1: {
-        state_type: "VIEW",
-        ref: "my-view",
-        transitions: {},
-      },
-    },
-  },
+        state_type: 'VIEW',
+        ref: 'my-view',
+        transitions: {}
+      }
+    }
+  }
+};
+
+const asyncNodeTest = async (resolvedValue: any, expectedActionType: string) => {
+  const plugin = new AsyncNodePlugin({
+    plugins: [new AsyncNodePluginPlugin()]
+  });
+
+  let deferredResolve: ((value: any) => void) | undefined;
+
+  let resolverInstance: Resolver;
+
+  plugin.hooks.onAsyncNode.tap('test', async (node: Node.Node) => {
+    return new Promise((resolve) => {
+      deferredResolve = resolve; // Promise would be resolved only once
+    });
+  });
+
+  let updateNumber = 0;
+
+  const player = new Player({ plugins: [plugin] });
+
+  player.hooks.viewController.tap('async-node-test', (vc) => {
+    vc.hooks.view.tap('async-node-test', (view) => {
+      view.hooks.onUpdate.tap('async-node-test', () => {
+        updateNumber++;
+      });
+    });
+  });
+
+  player.hooks.viewController.tap("async-node-test", (vc) => {
+    vc.hooks.view.tap("async-node-test", (view) => {
```

Comment:
67-68 is the same as 59-60. You don't need to tap it twice, you can accomplish all of this within one nested tap.

Reply:
the resolverInstance here as we've talked about doesn't accomplish what we want it to, which is to call the `onUpdate` hook, the `onUpdate` hook is exclusively called within the bigger scope of the view update process as the last step. 

this means if we want to make sure that the view is updated again through a 3rd manual update through `onUpdate` hook's counter, we need to update it using a `ViewInstance` that can be grabbed [here](https://github.com/player-ui/player/pull/389/files#diff-fbc47a3e301bdf595a6d686ded538fe6767e534af8821c421f97dc56264d1332R60) and calling update on that instance

Reply:
done